MobileDevProject/ViewController.swift

```diff
@@ -123,10 +124,22 @@ class ViewController: UIViewController, CLLocationManagerDelegate {
                 }
 
                 if isNavigating {
-                    if beacons[0].minor == navigatingBeacon.value && beacons[0].proximity == CLProximity.immediate {
+                    if beacons[0].minor == navigatingBeacon.value && beacons[0].proximity == CLProximity.near {
                         // we have arrived
                         isNavigating = false
-                        print("123223")
+                        // vibrate
+                        AudioServicesPlayAlertSound(SystemSoundID(kSystemSoundID_Vibrate))
+
+                        // show alert
+                        let alertController = UIAlertController(title: "Room Finder", message:
+                            "You have arrived!", preferredStyle: UIAlertControllerStyle.alert)
+                        alertController.addAction(UIAlertAction(title: "Dismiss", style: UIAlertActionStyle.default))
+                        self.present(alertController, animated: true, completion: nil)
+
+                        // clear circle and line
+                        self.lineShapeLayer.removeFromSuperlayer()
```

Comment:
May check if `lineShapeLayer` is not null. I will check it myself if we need that.